lib/value_range_check/dl_esm_inf/value_range_check.f90

```diff
@@ -89,31 +90,32 @@ subroutine DeclareField(this, name, value)
 
         implicit none
 
-        class(nan_test_PSyDataType), intent(inout), target :: this
+        class(value_range_check_psydatatype), intent(inout), target :: this
         character(*), intent(in) :: name
         type(r2d_field), intent(in) :: value
+        call this%PreDeclareVariable(name, value%data)
```

Comment:
Well, we call the base class, since the base class might do something useful.
In this case not much (it increases an index to know which variable index is current, and I think it can print some verbose/debug messages if required). Nothing that we urgently need here :)